### PR TITLE
fix: apply crate results from packing slip to crates during ship

### LINKED ISSUE #293: During ship, crates do not have the correct values set for `was_updated` & `result`
### Steps to reproduce
1. Lift a pallet with at least one crate that updates.
1. Verify that the packing slip shows the correct values for result and was_updated.
1. Ship and inspect the values after the pallets have been built.

### Expected behavior
`crate.was_updated()` => `True`
`crate.result` => `"Data updated successfully."`

### Actual behavior
`crate.was_updated()` => `False`
`crate.result` => `"This crate was never processed."`

This causes problems for Pallets that want to act only if crates were updated during lift. For example: https://github.com/agrc/warehouse/blob/master/sgid/AGOLPallet.py#L71

A possible fix would be to expand [`Pallet.add_packing_slip`](https://github.com/agrc/forklift/blob/cce42a1898224de373dd7f30ceeb89f4a67b8b27/src/forklift/models.py#L221) to loop through all of the crates in the slip and update these properties (are there any others?) of the actual crates. [Here](https://github.com/agrc/forklift/blob/cce42a1898224de373dd7f30ceeb89f4a67b8b27/src/forklift/engine.py#L595) is where that method is called in `engine.py`.

This is causing the AGOL pallet to not update data in AGOL since it never has updated crates during ship.

### PR DESCRIPTION
**Pull request by stdavis**
Fixes #293

Is this a bad idea?

### CODE REVIEW

File: tests/test_pallet.py [hunk 1]
```diff
@@ -345,3 +345,38 @@ def test_processing_time(self):
...
+            "is_ready_to_ship": True,
+            "requires_processing": True,
+            "message": "",
+            "crates": [
+                {
+                    "name": "fc5",
+                    "result": "Data updated successfully.",
+                    "crate_message": "",
+                    "message_level": "",
+                    "source": "c:\\users\\agrc-arcgis\\appdata\\local\\esri\\conda\\envs\\forklift\\lib\\site-packages\\forklift\\..\\forklift-garage\\sgid10.sde\\SGID10.BOUNDARIES.ZipCodes",
```
Comment (steveoh) [inline tests/test_pallet.py]: %localappdata% might be a better path to appdata\local

Comment (steveoh) [reply tests/test_pallet.py]: i don't think this works actually so... nevermind? Unless you want to research another solution

**Comment by stdavis** [comment]
image

File: tests/data/test_engine/packing-slip.json [hunk 2]
```diff
@@ -0,0 +1,80 @@
+[
+  {
+    "name": "c:\\forklift\\warehouse\\warehouse\\sgid\\AGOLPallet.py:AGOLPallet",
```
Comment (steveoh) [inline tests/data/test_engine/packing-slip.json]: do you want to make these paths generic?

**Comment by stdavis** [comment]
image

File: tests/data/test_engine/packing-slip.json (hunk 2 above)
Comment (stdavis) [reply tests/data/test_engine/packing-slip.json]: Oh yes. Thanks for the reminder...

File: tests/test_pallet.py (hunk 1 above)
Comment (stdavis) [reply tests/test_pallet.py]: The path doesn't really matter in this test.

**Pull request merged**